cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/CognitiveServiceBase.scala

```diff
@@ -154,11 +155,23 @@ trait HasAADToken extends HasServiceParams {
     setScalarParam(AADToken, v)
   }
 
-  def getAADToken: String = getScalarParam(AADToken)
-
-  def setAADTokenCol(v: String): this.type = setVectorParam(AADToken, v)
+  protected def refreshAADToken(token: String): String = {
+    if (SynapsePlatformUtils.isSynapseInternal) {
+      val expiryTime = SynapsePlatformUtils.getTokenExpiryTime(token)
+      val minutesLeft = (expiryTime.toInstant.toEpochMilli - System.currentTimeMillis()) / 60000
+      if (minutesLeft < 10) {
+        val newAADToken = SynapsePlatformUtils.getAccessTokenInternal()
+        setScalarParam(AADToken, newAADToken)
+        newAADToken
+      } else {
+        token
+      }
+    } else {
+      token
+    }
+  }
```

Comment:
It seems like this is hardwired for only using TokenLib which isnt really a generally used method for working with AAD tokens. Maybe in the future we can make the "refreshFunction" something settable or configurable in some way